scripts/import/import_EPMC.pl

```diff
@@ -993,9 +1005,253 @@ sub get_current_citations{
+
+      next unless !defined($publication);
+
+      my @feature_id;
+      if(defined $new_publications{$pmid}){
+        push(@{$new_publications{$pmid}}, $pheno_feat_id);
+      }
+      else{
+        push @feature_id, $pheno_feat_id;
+        $new_publications{$pmid} = \@feature_id;
+      }
+    }
+  }
+
+  foreach my $pmid (keys(%new_publications)){
+    my $feature_id_list = $new_publications{$pmid};
+
+    my $pub_data = get_epmc_data("search?query=ext_id:$pmid%20src:med");
+
+    if(defined $pub_data->{resultList}->{result}->{title}){
+      my $pub_title = $pub_data->{resultList}->{result}->{title};
+      my $pub_pmcid = $pub_data->{resultList}->{result}->{pmcid};
+      my $pub_author = $pub_data->{resultList}->{result}->{authorString};
+      my $pub_year = $pub_data->{resultList}->{result}->{pubYear};
+      my $pub_doi = $pub_data->{resultList}->{result}->{doi};
+
+      my @variant_list;
+      my $source_attrib_id;
+
+      foreach my $feature_id (@{$feature_id_list}){
+        # Get source and variant id
+        $pheno_sth->execute($feature_id)||die;
+        my $pheno_data = $pheno_sth->fetchall_arrayref();
+
+        next unless defined $pheno_data->[0]->[0];
+
+        my $source_id = $pheno_data->[0]->[0];
+        my $var_name = $pheno_data->[0]->[1];
+
+        my $v = $var_ad->fetch_by_name($var_name);
+        # Add variant if it exists - avoid undef values
+        if (defined $v){
+          push @variant_list, $v;
+        }
+
+        # Get source name from source table (ClinVar)
+        my $source_obj = $source_ad->fetch_by_dbID($source_id);
+        my $source_name = $source_obj->name();
+        $source_attrib_id = $citation_attribs->{$source_name};
+        die "No attrib of type 'citation_source' was found for '$source_name'!\n" unless defined $source_attrib_id;
+      }
+
+      # Clean title before insertion
+      if($pub_title =~ /^\[ ?[A-Za-z]{2}/){
+          $pub_title =~ s/^\[//;
+          $pub_title =~ s/\]//;
+      }
+
+      # Skip publication if title 'Not Available'
+      next if ($pub_title =~ /Not Available/);
+
+      ## create new object
+      my $publication = Bio::EnsEMBL::Variation::Publication->new( 
+                -title    => $pub_title,
+                -authors  => $pub_author,
+                -pmid     => $pmid,
+                -pmcid    => $pub_pmcid || undef,
+                -year     => $pub_year,
+                -doi      => $pub_doi,
+                -ucsc_id  => undef,
+                -variants => \@variant_list,
+                -adaptor  => $pub_ad
+                );
+
+      $pub_ad->store($publication,$source_attrib_id);
+    }
+  }
+}
+
+# Fetch all attribs that have type citation_source
+# Method used by type = 'phenotype'
+sub get_citation_attribs {
+  my $dba = shift;
+
+  my %attribs;
+  my $attrib_type_id;
+
+  my $stm_1 = $dba->dbc()->prepare(qq[ SELECT attrib_type_id FROM attrib_type WHERE code = 'citation_source' ]);
```

Comment:
This is only a style thing, but I find it far easier to work with meaningful names like '$attrib_ext_sth' rather than '$stm_1'